lighthouse-core/test/audits/preload-fonts-test.js

```diff
@@ -0,0 +1,214 @@
+    return {
+      devtoolsLogs: {[PreloadFontsAudit.DEFAULT_PASS]: networkRecordsToDevtoolsLog(networkRecords)},
+      URL: {finalUrl: 'https://example.com/foo/bar/page'},
+      CSSUsage: {stylesheets: [stylesheet]},
+    };
+  }
+
+  describe('font-display is optional', () => {
+    it('fails if the font is not preloaded', async () => {
+      stylesheet.content = `
+        @font-face {
+          font-display: optional;
+          src: url('/assets/font-a.woff');
+        }
+      `;
+
+      networkRecords = [
+        {
+          url: 'https://example.com/assets/font-a.woff',
+          endTime: 3, startTime: 1,
+          resourceType: 'Font',
+          isLinkPreload: false,
+        },
+      ];
+
+      const result = await PreloadFontsAudit.audit(getArtifacts(), context);
+      const items = [
+        {url: networkRecords[0].url},
+      ];
+      expect(result.score).toEqual(0);
+      expect(result.details.items).toEqual(items);
+    });
+
+    it('passes if the font is preloaded', async () => {
+      stylesheet.content = `
+        @font-face {
+          font-display: optional;
+          src: url('/assets/font-a.woff');
+        }
+      `;
+
+      networkRecords = [
+        {
+          url: 'https://example.com/assets/font-a.woff',
+          endTime: 3, startTime: 1,
+          resourceType: 'Font',
+          isLinkPreload: true,
+        },
+      ];
+
+      const result = await PreloadFontsAudit.audit(getArtifacts(), context);
+      expect(result.details.items).toEqual([]);
+      expect(result.score).toEqual(1);
+    });
+  });
+
+  describe('font-display is not optional', () => {
+    it('passes if the font is not preloaded', async () => {
+      stylesheet.content = `
+        @font-face {
+          src: url('/assets/font-a.woff');
+        }
+      `;
+
+      networkRecords = [
+        {
+          url: 'https://example.com/assets/font-a.woff',
+          endTime: 3, startTime: 1,
+          resourceType: 'Font',
+          isLinkPreload: false,
+        },
+      ];
+
+      const result = await PreloadFontsAudit.audit(getArtifacts(), context);
+      expect(result.score).toEqual(1);
+      expect(result.details.items).toEqual([]);
+      expect(result.notApplicable).toEqual(true);
+    });
+
+    it('passes if the font is preloaded', async () => {
+      stylesheet.content = `
+        @font-face {
+          src: url('/assets/font-a.woff');
+        }
+      `;
+
+      networkRecords = [
+        {
+          url: 'https://example.com/assets/font-a.woff',
+          endTime: 3, startTime: 1,
+          resourceType: 'Font',
+          isLinkPreload: true,
+        },
+      ];
+
+      const result = await PreloadFontsAudit.audit(getArtifacts(), context);
+      expect(result.score).toEqual(1);
+      expect(result.details.items).toEqual([]);
+      expect(result.notApplicable).toEqual(true);
+    });
+  });
+
+  it('is not applicable on fonts where font-display is not optional', async () => {
```

Comment:
```suggestion
  it('ignores fonts where font-display is not optional', async () => {
```

(not applicable conflates with the scoring method of an audit)

Reply:
I'll change that! 

For my clarity, why is that a bad thing? i.e. I currently check `expect(result.notApplicable).toEqual(true);` in this test

Reply:
Oh, I missed that check. nvm :)